### PR TITLE
feat: band registration api

### PR DESCRIPTION
**Pull request by kanatsanan6**
## Endpoint

- `GET http://127.0.0.1:5001/loma-nkaf/us-central1/api/bands`
  - get all bands by userId

    | params | require | description |
    |--------|---------|-------------|
    | userId | yes     | user id     |

- `POST http://127.0.0.1:5001/loma-nkaf/us-central1/api/bands`
  - create band
 
    | params                           | require | type                                                    |
    |----------------------------------|---------|---------------------------------------------------------|
    | userId                           | yes     | string                                                  |
    | bandName                         | yes     | string                                                  |
    | firstPromotedSong                | no      | string                                                  |
    | secondPromotedSong               | no      | string                                                  |
    | socialMedia['facebook']          | no      | string                                                  |
    | socialMedia['instagram']         | no      | string                                                  |
    | socialMedia['tiktok']            | no      | string                                                  |
    | socialMedia['website']           | no      | string                                                  |
    | streamingPlatform['spotify']     | no      | string                                                  |
    | streamingPlatform['youtube']     | no      | string                                                  |
    | streamingPlatform['apple_music'] | no      | string                                                  |
    | lineMelody                       | no      | string                                                  |
    | songRequest                      | no      | boolean (default: false if send null)                   |
    | description                      | no      | string                                                  |
    | lineBeacon                       | no      | Array of {hardware_id: hardware_id, passcode: passcode} |
    | bandImage                        | no      | string                                                  |
    | qrImage                          | no      | string                                                  |

## TODO

- Add token verification middleware (next PR)

### CODE REVIEW

File: functions/src/api/controllers/bands/bandController.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      socialMedia: req.body.socialMedia || {},
+      streamingPlatform: req.body.streamingPlatform || {},
+      lineMelody: req.body.lineMelody || null,
+      songRequest: req.body.songRequest || false,
+      description: req.body.description || null,
+      lineBeacon: req.body.lineBeacon || [],
+    };
+
+    // TODO: hard-coded for now
+    const bucketName = "loma-nkaf";
```
Comment (Supermalonz) [inline functions/src/api/controllers/bands/bandController.ts:51]: 🤔

Comment (Supermalonz) [reply functions/src/api/controllers/bands/bandController.ts:51]: I see hard-coded

Comment (kanatsanan6) [reply functions/src/api/controllers/bands/bandController.ts:51]: 🤫

Comment (kanatsanan6) [reply functions/src/api/controllers/bands/bandController.ts:51]: Just wanna point out these to a test bucket, I will set into an env afterward na

**Review by patorseing — APPROVED** [review]
Your code is quite nice na, but don't forget about hard code.

**Pull request merged**